haddock-api/src/Haddock/Types.hs

```diff
@@ -217,6 +217,9 @@ data ExportItem name
         -- | A declaration.
         expItemDecl :: !(LHsDecl name)
 
+        -- | Bundled patterns for a data type declaration
+      , expItemPats :: ![(HsDecl name,DocForDecl name,[(name, Fixity)])]
```

Comment:
I think we should just reuse the already present `expItemFixities` field. Also I presume with your current design fixities won't be saved to interface files (`toInstalledInterface` etc.).